packages/hoppscotch-common/src/pages/index.vue

```diff
@@ -203,28 +211,39 @@ const onSaveModalClose = () => {
   }
 }
 
-watch(confirmSync, (newValue) => {
-  if (newValue) {
-    toast.show(t("confirm.sync"), {
-      duration: 0,
-      action: [
-        {
-          text: `${t("action.yes")}`,
-          onClick: (_, toastObject) => {
-            syncTabState()
-            toastObject.goAway(0)
+watch(
+  () => confirmSync.value.initialSync,
+  (newValue) => {
+    if (newValue) {
+      toast.show(t("confirm.sync"), {
+        duration: 0,
+        action: [
+          {
+            text: `${t("action.yes")}`,
+            onClick: (_, toastObject) => {
+              syncTabState()
+              changeCurrentSyncStatus({
+                initialSync: true,
+                sync: true,
+              })
+              toastObject.goAway(0)
+            },
           },
-        },
-        {
-          text: `${t("action.no")}`,
-          onClick: (_, toastObject) => {
-            toastObject.goAway(0)
+          {
+            text: `${t("action.no")}`,
+            onClick: (_, toastObject) => {
+              changeCurrentSyncStatus({
+                initialSync: true,
+                sync: false,
+              })
+              toastObject.goAway(0)
+            },
           },
-        },
-      ],
-    })
+        ],
+      })
+    }
   }
-})
+)
```

Comment:
We don't need to do it in a watcher, because the only place we're triggering this is in the `onLoggedIn` hook. ( see the comment on `onLoggedIn`  )

```suggestion
const showSyncToast = () => {
  toast.show(t("confirm.sync"), {
    duration: 0,
    action: [
      {
        text: `${t("action.yes")}`,
        onClick: (_, toastObject) => {
          syncTabState()
          changeCurrentSyncStatus({
            initialSync: true,
            sync: true,
          })
          toastObject.goAway(0)
        },
      },
      {
        text: `${t("action.no")}`,
        onClick: (_, toastObject) => {
          changeCurrentSyncStatus({
            initialSync: true,
            sync: false,
          })
          toastObject.goAway(0)
        },
      },
    ],
  })
}
```